api/src/main/java/io/strimzi/api/kafka/model/listener/v2/GenericKafkaListenerConfiguration.java

```diff
@@ -0,0 +1,168 @@
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Buildable(
+    editableEnabled = false,
+    builderPackage = Constants.FABRIC8_KUBERNETES_API
+)
+@EqualsAndHashCode
+public class GenericKafkaListenerConfiguration implements Serializable, UnknownPropertyPreserving {
+    private static final long serialVersionUID = 1L;
+
+    private CertAndKeySecretSource brokerCertChainAndKey;
+    private String ingressClass;
+    private NodeAddressType preferredAddressType;
+    private ExternalTrafficPolicy externalTrafficPolicy;
+    private List<String> loadBalancerSourceRanges = new ArrayList<>(0);
+    private Boolean useServiceDnsDomain;
+    private GenericKafkaListenerConfigurationBootstrap bootstrap;
+    private List<GenericKafkaListenerConfigurationBroker> brokers;
+
+    private Map<String, Object> additionalProperties = new HashMap<>(0);
+
+    @Description("Reference to the `Secret` which holds the certificate and private key pair which will be used for this listener. " +
+            "The certificate can optionally contain the whole chain. " +
+            "This field can be used only with listeners with enabled TLS encryption.")
+    public CertAndKeySecretSource getBrokerCertChainAndKey() {
+        return brokerCertChainAndKey;
+    }
+
+    public void setBrokerCertChainAndKey(CertAndKeySecretSource brokerCertChainAndKey) {
+        this.brokerCertChainAndKey = brokerCertChainAndKey;
+    }
+
+    @Description("Configures the `Ingress` class that defines which `Ingress` controller will be used. " +
+            "If not set, the `Ingress` class is set to `nginx`. " +
+            "This field can be used only with `ingress` type listener.")
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @JsonProperty("class")
+    public String getIngressClass() {
+        return ingressClass;
+    }
+
+    public void setIngressClass(String ingressClass) {
+        this.ingressClass = ingressClass;
+    }
+
+    @Description("Defines which address type should be used as the node address. " +
+            "Available types are: `ExternalDNS`, `ExternalIP`, `InternalDNS`, `InternalIP` and `Hostname`. " +
+            "By default, the addresses will be used in the following order (the first one found will be used):\n" +
+            "* `ExternalDNS`\n" +
+            "* `ExternalIP`\n" +
+            "* `InternalDNS`\n" +
+            "* `InternalIP`\n" +
+            "* `Hostname`\n" +
+            "\n" +
+            "This field can be used to select the address type which will be used as the preferred type and checked first. " +
+            "In case no address will be found for this address type, the other types will be used in the default order." +
+            "This field can be used only with `nodeport` type listener.")
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    public NodeAddressType getPreferredAddressType() {
```

Comment:
I called `ingressClass` that way because the field it changes is called like that and I do not think other things have any similar concept. On the other hand, I can for example imagine that `preferredAddressType` is in the future also used for load balancers to define whether a DNS name or an IP address should be preferred.

If I had to choose, I would probably tend to rename the `ingressClass` to class rather than the other way around. WDYT?

Reply:
Yeah, it's tricky. One of the hazards of using an object with properties which have to cope with the union of the different types. I think `class` would be a bit weird, because it would configure the `ingressClass` field you mentioned in the case of ingress and something else for another type so there wouldn't be a 1-to-1 correspondence. Not that there has to be, but it's helpful for people's mental model when there is. I think I would prefer what you already have over using `class`.

If load balancers gained a similar thing then having both `preferredNodeportAddressType` and `preferredLoadBalancerAddressType` would not be terrible. But I'm not wedded to the idea.

Reply:
I will change it to `preferredNodePortAddressType` since the load balancers would have anyway different implementation if we add them in the future.